Alerting Docs: Update Alert rule and Notification Templating (#94069)
grafana/grafana: docs/sources/shared/alerts/template-language.md

```

This is useful when writing reusable templates. For example, to print all alerts you might write the following:

```go
{{ template "print_alerts" .Alerts }}
```

Then to print just the firing alerts you could write this:

```go
{{ template "print_alerts" .Alerts.Firing }}
```

This works because both `.Alerts` and `.Alerts.Firing` is a list of alerts.

```go
{{ define "print_alerts" }}
{{ range . }}
{{ template "print_labels" .SortedLabels }}
{{ end }}
{{ end }}
```

{{% admonition type="note" %}}
You cannot create independent, reusable templates for labels and annotations as you can with notification templates. In alert rule templates, you need to write each template inline within the label or annotation field.
{{% /admonition %}}

## Comments

You can add comments with `{{/*` and `*/}}`:

```go
{{/* This is a comment */}}
```

To avoid adding line breaks, use:

```go
{{- /* This is a comment with no leading or trailing line breaks */ -}}
```

## Indentation

You can use indentation, both tabs and spaces, and line breaks, to make templates more readable:

```go
{{ range .Alerts }}
  {{ range .Labels.SortedPairs }}
    {{ .Name }} = {{ .Value }}
  {{ end }}
{{ end }}
```

> However, indentation in the template is also present in the text. Next, we look at how to remove it.

Reviewer: ```suggestion
However, indentation in the template is also present in the text. 
```

### Remove spaces and line breaks

In text/template use `{{-` and `-}}` to remove leading and trailing spaces and line breaks.

For example, when using indentation and line breaks to make a template more readable:

```go
{{ range .Alerts }}
  {{ range .Labels.SortedPairs }}
    {{ .Name }} = {{ .Value }}
  {{ end }}
{{ end }}
```

> The indentation and line breaks will also be present in the text:

Reviewer: ```suggestion
The indentation and line breaks are also present in the text:
```
